rest/model/task.go

```diff
@@ -51,6 +52,7 @@ type APITask struct {
 	Artifacts          []APIFile        `json:"artifacts"`
 	DisplayOnly        bool             `json:"display_only"`
 	ExecutionTasks     []APIString      `json:"execution_tasks,omitempty"`
+	Mainline           bool             `json:"mainline"`
```

Comment:
Given the the sources of evergreen versions we've been adding I think this is ambiguous and will become more ambiguous. I would do the inverse and call it IsPatch, which will be if it's a patch or pull request

Reply:
I just think that the evergreen source is probably not the way we want to think about it, and the whole point of having the rest models is to be able to detach users from the experience of our database.

Reply:
I meant from a conceptual standpoint. Like is a task from a triggered commit mainline or not? I feel like "this is a patch" is the most unambiguous

Reply:
What about commit queue, API triggered builds or pull requests.

Reply:
Commit queue and pull requests are patches

Reply:
Yes, though I think "mainline" is a more cohesive set than "patch"

Reply:
Given that we might have additional types of requesters in the future, I think this should pass it through:

```
	// version requester types
	PatchVersionRequester       = "patch_request"
	GithubPRRequester           = "github_pull_request"
	RepotrackerVersionRequester = "gitter_request"
	TriggerRequester            = "trigger_request"
	MergeTestRequester          = "merge_test"
	AdHocRequester              = "ad_hoc"
```

Reply:
I don't think we should expose this implementation detail to users.